data/data/openstack/masters/main.tf

```diff
@@ -147,7 +155,7 @@ resource "openstack_compute_instance_v2" "master_conf_1" {
   flavor_id = data.openstack_compute_flavor_v2.masters_flavor.id
   image_id = var.openstack_master_root_volume_size == null ? data.openstack_images_image_v2.base_image.id : null
   security_groups = local.master_sg_ids
-  availability_zone = var.openstack_master_availability_zones[1 % length(var.openstack_master_availability_zones)]
+  availability_zone = var.openstack_master_availability_zones[1]
```

Comment:
We used to pass a slice with the declared zones. Terraform was responsible for distributing them.

Now tfvars builds a slice of length(number-of-masters) containing the zones ready to be used, so Terraform just has to pick the one corresponding to the current instance.